ext/CatalystStructuralIdentifiabilityExtension/structural_identifiability_extension.jl

```diff
@@ -0,0 +1,144 @@
+### Structural Identifiability ODE Creation ###
+
+# For a reaction system, list of measured quantities and known parameters, generate a StructuralIdentifiability compatible ODE. 
+"""
+    si_ode(rs::ReactionSystem; measured_quantities=observed(rs), known_p = Num[], ignore_no_measured_warn=false)
+
+Creates a ODE system of the form used within the StructuralIdentifiability.jl package. The output system is compatible with all StructuralIdentifiability functions. 
+
+Arguments:
+- `rs::ReactionSystem`; The reaction system we wish to convert to an ODE.
+- `measured_quantities=observed(rs)`: The quantities of the system we can measure. May either be equations (e.g. `x1 + x2`), or single species (e.g. `:x`). Defaults to the systems observables.
+- `known_p = Num[]`: List of parameters which values are known. 
+- `ignore_no_measured_warn=false`: If set to `true`, no warning is provided when the `measured_quantities` vector is empty. 
+
+Example:
+```julia
+rs = @reaction_network begin
+    (p,d), 0 <--> X
+end
+si_ode(rs; measured_quantities = [:X], known_p = [:p])
+
+Notes:
+This function is part of the StructuralIdentifiability.jl extension. StructuralIdentifiability.jl must be imported to access it.
+```
+"""
+function Catalyst.make_si_ode(rs::ReactionSystem; measured_quantities = [], known_p = [], 
+                              ignore_no_measured_warn=false, remove_conserved = true)
+    # Creates a MTK ODESystem, and a list of measured quantities (there are equations).
+    # Gives these to SI to create an SI ode model of its preferred form.
+    osys, conseqs, _ = make_osys(rs; remove_conserved)
+    measured_quantities = make_measured_quantities(rs, measured_quantities, known_p, conseqs; ignore_no_measured_warn)
+    return SI.preprocess_ode(osys, measured_quantities)[1]
+end
+
+### Structural Identifiability Wrappers ###
+
+# Creates dispatch for SI's local identifiability analysis function.
+function SI.assess_local_identifiability(rs::ReactionSystem, args...; measured_quantities = Num[], 
```

Comment:
Shouldn't this and all the other `assess_*` functions get a doc string too, explaining `args` and `kwargs` are forwarded to SI and provide a way to pass additional SI-specific parameters?

Reply:
It will show both I believe, but yours will have 
```julia
    SI.assess_local_identifiability(rs::ReactionSystem, args...; ...)
```
as the function argument, so it is clear it is the method (i.e. version) that takes `ReactionSystem`s.

Reply:
Try it on `convert` when you've got Catalyst loaded (`convert` is a base Julia function).

Reply:
Got, thanks. Will add docstrings then.

Should these be added to an extensions API section as well?

Reply:
added